src/puppet.ts

```diff
@@ -257,17 +258,18 @@ export abstract class Puppet extends EventEmitter {
    *
    *
    */
-  public emit (event: 'dong',       data?: string)                                                         : boolean
-  public emit (event: 'error',      error: Error)                                                          : boolean
-  public emit (event: 'friendship', friendshipId: string)                                                  : boolean
-  public emit (event: 'login',      contactId: string)                                                     : boolean
-  public emit (event: 'logout',     contactId: string)                                                     : boolean
-  public emit (event: 'message',    messageId: string)                                                     : boolean
-  public emit (event: 'reset',      reason: string)                                                        : boolean
-  public emit (event: 'room-join',  roomId: string, inviteeIdList: string[],  inviterId: string)           : boolean
-  public emit (event: 'room-leave', roomId: string, leaverIdList: string[], remover?: string)              : boolean
-  public emit (event: 'room-topic', roomId: string, newTopic: string, oldTopic: string, changerId: string) : boolean
-  public emit (event: 'scan',       qrcode: string, status: number, data?: string)                         : boolean
+  public emit (event: 'dong',        data?: string)                                                         : boolean
+  public emit (event: 'error',       error: Error)                                                          : boolean
+  public emit (event: 'friendship',  friendshipId: string)                                                  : boolean
+  public emit (event: 'login',       contactId: string)                                                     : boolean
+  public emit (event: 'logout',      contactId: string)                                                     : boolean
+  public emit (event: 'message',     messageId: string)                                                     : boolean
+  public emit (event: 'reset',       reason: string)                                                        : boolean
+  public emit (event: 'room-join',   roomId: string, inviteeIdList: string[],  inviterId: string)           : boolean
+  public emit (event: 'room-leave',  roomId: string, leaverIdList: string[], remover?: string)              : boolean
+  public emit (event: 'room-topic',  roomId: string, newTopic: string, oldTopic: string, changerId: string) : boolean
+  public emit (event: 'room-invite', roomInvitation: RoomInvitation)                                        : boolean
```

Comment:
Please leave other `emit()` lines to be un-modified in your PR, to make it clear.